packages/alfa-css/src/value/contain.ts

```diff
@@ -0,0 +1,237 @@
+import type { Hash } from "@siteimprove/alfa-hash";
+import { Err, Result } from "@siteimprove/alfa-result";
+import { Parser } from "@siteimprove/alfa-parser";
+
+import { Keyword } from "./keyword.js";
+import type { Resolvable } from "./resolvable.js";
+import { Value } from "./value.js";
+
+import { Token, type Parser as CSSParser } from "../syntax/index.js";
+import { Slice } from "@siteimprove/alfa-slice";
+
+const { either } = Parser;
+
+/**
+ * {@link https://developer.mozilla.org/en-US/docs/Web/CSS/contain#formal_syntax}
+ *
+ * @public
+ */
+export type Contain =
+  | Keyword<"none">
+  | Keyword<"strict">
+  | Keyword<"content">
+  | ContainFlags;
+
+/**
+ * @public
+ */
+export class ContainFlags
+  extends Value<"contain-flags", false>
+  implements Resolvable<ContainFlags, never>
+{
+  public static of(
+    size: boolean,
+    inlineSize: boolean,
+    layout: boolean,
+    style: boolean,
+    paint: boolean,
+  ) {
+    return new ContainFlags(size, inlineSize, layout, style, paint);
+  }
+
+  private readonly _size: boolean;
+  private readonly _inlineSize: boolean;
+  private readonly _layout: boolean;
+  private readonly _style: boolean;
+  private readonly _paint: boolean;
```

Comment:
**Question:** Would it make sense to store that as an `alfa-flag` 🤔 ? (see also how the DOM `Node` names its flags)  We'd still need a wrapper class as it needs to extends `Value`, so maybe that wouldn't help at all. But that could help in providing the query and build helpers.

Reply:
It could make sense, but I think I'll leave it as it is for now. 😄